include/gadget/buffer.h

```diff
@@ -18,6 +18,7 @@
 	} name SEC(".maps");				\
 	const void *gadget_map_tracer_##name __attribute__((unused));
 
+#ifndef GADGET_NO_RESERVE
```

Comment:
I'm wondering about this API. It seems it's complicated for users having another knob to set. 

I'm wondering if we can control all of it by using `MAX_EVENT_SIZE`, we can define it to be 1 by default (very small overhead) and then gadget developers can change it based on their specific event:

```c
#define MAX_EVENT_SIZE (sizeof(struct event))
#include <gadget/buffer.h>
```